tumfoodery/tripster
# Apollo Server Setup

# Description
- Creates Apollo Server
  - Schema
  - Resolvers
  - Firebase Datasource
- Supports `isLoggedIn` Query 🍔 
- Supports `login` and `signup` Mutation 🍄


## Test with Graphiql Playground
1. `npm start`
2. [localhost:4000](http://localhost:4000)
3. Paste this query:
```
mutation SignupUser($email: String!, $password: String!) {
  signup(email: $email, password: $password)
}
```
4. Paste these Query Variables
```
{
  "email": "iam@mikey.baby",
  "password": "tester"
}
```
5. Click the play button
6. You should see this response:
```
{
  "data": {
    "signup": "The email address is already in use by another account."
  }
}
```
This is because I've already signed up. You can change the email value to your own and then the response value with be the user uid.

## To Do
- [x] Refactor .env

## Questions
- [x] Should the firebaseConfig be passed in through the constructor or used directly in the firebase datasource?
- **Answer:** Passed through contructor, but assembled in index with env.
